frontend/src/pages/WorkloadDetails/WorkloadPodLogs.tsx

```diff
@@ -730,65 +763,60 @@ export class WorkloadPodLogs extends React.Component<WorkloadPodLogsProps, Workl
       });
   };
 
-  private doShowAndHide = () => {
-    this.filterEntries(this.state.entries, this.state.showLogValue, this.state.hideLogValue);
-    this.setState({
-      entries: [...this.state.entries],
-      showClearShowLogButton: !!this.state.showLogValue,
-      showClearHideLogButton: !!this.state.hideLogValue
-    });
-  };
-
   private checkSubmitShow = event => {
     const keyCode = event.keyCode ? event.keyCode : event.which;
     if (keyCode === RETURN_KEY_CODE) {
       event.preventDefault();
-      this.doShowAndHide();
+      this.setState({
+        showClearShowLogButton: !!event.target.value,
+        showLogValue: event.target.value
+      });
     }
   };
 
   private updateShow = val => {
     if ('' === val) {
       this.clearShow();
-    } else {
-      this.setState({ showLogValue: val });
     }
   };
 
-  private filterEntries = (entries: Entry[], showValue: string, hideValue: string): void => {
-    entries.forEach(e => (e.isHidden = undefined));
+  private filteredEntries = memoize((entries: Entry[], showValue: string, hideValue: string, useRegex: boolean) => {
```

Comment:
I think a comment about what `memoize` is doing could be helpfule here.  I had to go look up what this library was doing and I still may not be sure.  I guess what this is about is caching the the filteredEntries for a combination of entries/showValue/hideValue/useRegex ?  That cache could get large if the user makes a lot of changes to the params and the set is large, no?

Reply:
I somehow though that memoization was a widespread concept (there is, even, a [Wikipedia article about it](https://en.wikipedia.org/wiki/Memoization)). In typescript/javascript I think is somewhat common.

Anyway, you are right that it is caching the result given a set of arguments. This specific library has a single element cache by default (see: https://www.npmjs.com/package/micro-memoize/v/4.0.9#maxsize), so I think this should be OK.